File: frontend/src/types/entities.ts
export type CategoryType = 'PERSON' | 'LOCATION' | 'ORGANIZATION' | 'CONCEPT';

export interface Fact {
  book_part_id: string;
  content: string;
  occurrences: number;
  sibling_index: number | null;
  sibling_total: number | null;
}

export interface EntityResponseSchema {
  name: string;
  alternative_names: string[];
  category: string;
  facts: Fact[];
}

File: backend/schemas/entities.py
from typing import List, Literal
import uuid
from pydantic import BaseModel

CategoryType = Literal['PERSON', 'LOCATION', 'ORGANIZATION', 'CONCEPT']


class Fact(BaseModel):
    book_part_id: uuid.UUID
    content: str
    occurrences: int
    sibling_index: int | None
    sibling_total: int | None


class EntityResponseSchema(BaseModel):
    name: str
    alternative_names: List[str]
    category: str
    facts: List[Fact]
